learn2track/models/ffnn_classification.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+import smartlearner.initializers as initer
+import theano
+import theano.tensor as T
+from learn2track.models import FFNN
+from learn2track.models.layers import LayerDense
+from smartlearner.interfaces.loss import Loss
+
+from learn2track.utils import l2distance
+
+floatX = theano.config.floatX
+
+
+class FFNN_Classification(FFNN):
+    """ A standard FFNN model with a classification (sigmoid) layer stacked on top of it.
+    """
+
+    def __init__(self, volume_manager, input_size, hidden_sizes, **_):
+        """
+        Parameters
+        ----------
+        volume_manager : :class:`VolumeManger` object
+            Use to evaluate the diffusion signal at specific coordinates.
+        input_size : int
+            Number of units each element X has.
+        hidden_sizes : int, list of int
+            Number of hidden units each FFNN layer should have.
+        """
+        super().__init__(input_size, hidden_sizes)
+        self.volume_manager = volume_manager
+        self.output_size = 1  # Positive class probability
+
+        self.layer_classification = LayerDense(self.hidden_sizes[-1], self.output_size, activation="sigmoid")
+
+    def initialize(self, weights_initializer=initer.UniformInitializer(1234)):
+        super().initialize(weights_initializer)
+        self.layer_classification.initialize(weights_initializer)
+
+    @property
+    def hyperparameters(self):
+        hyperparameters = super().hyperparameters
+        return hyperparameters
+
+    @property
+    def parameters(self):
+        return super().parameters + self.layer_classification.parameters
+
+    def _fprop(self, Xi, *args):
+        # Xi.shape : (batch_size, 4)
+        # coords + dwi ID
+
+        # coords : brain 3D coordinates.
+        # coords.shape : (batch_size, 4) where the last column is a dwi ID.
+        # args.shape : n_layers * (batch_size, layer_size)
+        coords = Xi[:, :4]
+
+        # Get diffusion data.
+        # data_at_coords.shape : (batch_size, input_size)
+        data_at_coords = self.volume_manager.eval_at_coords(coords)
```

Comment:
Might be overkill to use trilinear interpolation since coords are all integer. That said, with that the model could be trained to do super-resolution, i.e. by providing coords that are in between voxels (in a sense, this is equivalent to having smaller voxels sizes, thus better resolution).